### PR TITLE
Add REST API documentation for WooCommerce Stripe.

### PR DESCRIPTION
**Pull request by gedex**
This is a draft version that has a status request for feedback.

Available endpoints:

* `/stripe/settings` &mdash; View and update (single or batch). The endpoints, requests, and responses have a similar structure with [WooCommerce core setting options REST API](https://woocommerce.github.io/woocommerce-rest-api-docs/?shell#setting-options).
* Orders? For example: given an order ID the REST API allows to retrieve Stripe-related information (e.g. customer ID, card ID, etc). Most of them are stored as order meta so that can be handled by WC REST API too by whitelisting some meta.

/cc @allendav @budzanowski 

EDIT: We'll be using https://woocommerce.github.io/woocommerce-rest-api-docs/?shell#payment-gateways. PR is updated to give usage example.

### CODE REVIEW

File: docs/api/settings.md [hunk 1]
```diff
@@ -0,0 +1,283 @@
+REST API: Settings
+==================
+
+The settings REST API allows you to view and update WooCommerce Stripe settinngs.
```
Comment (budzanowski) [inline docs/api/settings.md]: settinngs typo

File: docs/api/settings.md [hunk 2]
```diff
@@ -0,0 +1,283 @@
...
+The endpoints, requests, and responses have similar structure with [WooCommerce core setting options REST API](https://woocommerce.github.io/woocommerce-rest-api-docs/?shell#setting-options).
```
Comment (budzanowski) [inline docs/api/settings.md]: I would say "share a similar structure with" or "have a similar structure to" but I am not English native - it just sounds better. Not very important suggestion.

File: docs/api/settings.md [hunk 3]
```diff
@@ -0,0 +1,283 @@
...
+
+## Setting option properties
+
+| Attribute     | Type   | Description                           |             |
```
Comment (budzanowski) [inline docs/api/settings.md]: I would say that it is a Name and not Attribute, The last comment misses a label - maybe _Access Type_

File: docs/api/settings.md [hunk 4]
```diff
@@ -0,0 +1,283 @@
...
+| ------------- | ------ | ------------------------------------- | ----------- |
+| `id`          | string | A unique identifier for the setting.  | `READ-ONLY` |
+| `label`       | string | A human readable label for the setting used in interfaces. | `READ-ONLY` |
+| `description` | string | A human readable description for the setting used in interfaces. | `READ-ONLY` |
+| `value`       | mixed  | Setting value. | |
```
Comment (budzanowski) [inline docs/api/settings.md]: The last column is empty.

File: docs/api/settings.md (hunk 1 above)
Comment (gedex) [inline docs/api/settings.md]: Fixed in https://github.com/woocommerce/woocommerce-gateway-stripe/pull/314/commits/7abd2c6e22a15feadfdfed0282db71601a872bd4.

File: docs/api/settings.md (hunk 2 above)
Comment (gedex) [inline docs/api/settings.md]: I like your suggestion.

Comment (gedex) [inline docs/api/settings.md]: Updated in https://github.com/woocommerce/woocommerce-gateway-stripe/pull/314/commits/cb9abcf6da20b8586dd58f04cce636294119adf2.

File: docs/api/settings.md (hunk 4 above)
Comment (gedex) [inline docs/api/settings.md]: It's intended to be not `READ-ONLY`. Maybe put `WRITABLE`?

File: docs/api/settings.md (hunk 3 above)
Comment (gedex) [inline docs/api/settings.md]: 👍

File: docs/api/settings.md (hunk 4 above)
Comment (gedex) [inline docs/api/settings.md]: or `EDITABLE`.

File: docs/api/settings.md (hunk 3 above)
Comment (gedex) [inline docs/api/settings.md]: Updated in https://github.com/woocommerce/woocommerce-gateway-stripe/pull/314/commits/ef0dabd19bb3b9c764763ec6dc4a3252b33555de.

File: docs/api/settings.md (hunk 4 above)
Comment (budzanowski) [inline docs/api/settings.md]: Yes, they would both work. What about `READ-WRITE'?

**Comment by gedex** [comment]
@budzanowski @mattyza I updated the doc to reflect the usage of [payment gateways API from WC core](https://woocommerce.github.io/woocommerce-rest-api-docs/?shell#payment-gateways). Since we don't expose custom endpoints for Stripe, I only added this documentation as README.md in `docs/api/` so that's rendered instantly when browsed from GitHub. The documentation explains how payment gateways API can be used to manage WooCommerce Stripe settings.

Preview: https://github.com/woocommerce/woocommerce-gateway-stripe/blob/ba852ce6f97021248370834022682aae6bc1eb69/docs/api/README.md.

**Comment by budzanowski** [comment]
Good to go! :shipit:

**Comment by allendav** [comment]
Sorry I missed this. I will read it promptly.

**Pull request merged**